### ISSUE TITLE
Support for logstash 6.x and 7.x needed

### ISSUE THREAD

**Issue by BillGrim** [connamara/logstash-filter-fix_protocol#87]
I trying to upgrade ELK 5.5.2 to ELK 6.8 or ELK 7.1, but get error messages shown below when running `./bin/logstash -t -f config/logstash.conf` with logstash 7.1.
error message:
```
[2019-06-18T17:59:33,192][ERROR][logstash.plugins.registry] Tried to load a plugin's code, but failed. {:exception=>#<LoadError: no such file to load -- i18n/core_ext/string/interpolate>, :path=>"logstash/filters/fix_protocol", :type=>"filter", :name=>"fix_protocol"}
[2019-06-18T17:59:33,202][FATAL][logstash.runner          ] The given configuration is invalid. Reason: Couldn't find any filter plugin named 'fix_protocol'. Are you sure this is correct? Trying to load the fix_protocol filter plugin resulted in this error: no such file to load -- i18n/core_ext/string/interpolate
[2019-06-18T17:59:33,211][ERROR][org.logstash.Logstash    ] java.lang.IllegalStateException: Logstash stopped processing because of an error: (SystemExit) exit
```
I've already installed fix_protocol plugin with  `./bin/logstash-plugin install logstash-filter-fix_protocol`, and got result `Validating logstash-filter-fix_protocol
Installing logstash-filter-fix_protocol
Installation successful`

So I think it is a problem related to version mis-compatibility.

I'd appreciate it if anyone can add a pull request to this plugin. Thanks.

**Comment by nktl** [comment]
+1, same problem here

**Comment by zandabanda** [comment]
+1

**Comment by daino3** [comment]
@cbusbey - you guys want to take ownership of this project? Unlikely I'll be maintaining this going forward.

**Comment by cbusbey** [comment]
Of course.  We would gladly merge a pr from the community to address this issue.

**Comment by ronyv** [comment]
Getting the same issue on logstash version 7.3. It was working fine with version 6.6.0. 
Any updates on this issue ???
Is there any other FIX message parser plugins available for Logstash ?

Any help on this will be greatly appreciated!

**Comment by nktl** [comment]
We are very desperate to get it up and running with modern ELK too, at the moment simply considering running separate logstash 6.6 instance just for this thing....

**Comment by lechaimwilson** [comment]
+1

**Comment by zandabanda** [comment]
Could this be related to jruby 9.2.5.0 -> 9.2.6.0 (see Logstash 6.7.0 release notes:  https://www.elastic.co/guide/en/logstash/6.8/logstash-6-7-0.html)

https://github.com/elastic/logstash/pull/10425/files

Assuming this breaks in going from Logstash 6.6 -> 6.7

**Comment by ourcookiemonster** [comment]
The error we see when trying to use the plugin in logstash-7.4.2 is the following:

[2019-11-15T16:39:20,984][ERROR][logstash.plugins.registry] Tried to load a plugin's code, but failed. {:exception=>#<LoadError: no such file to load -- i18n/core_ext/string/interpolate>, :path=>"logstash/filters/fix_protocol", :type=>"filter", :name=>"fix_protocol"}


It is not clear what the plug-in's dependency to i18n is since we were able to workaround the issue by simply placing an empty file in this location:
logstash-7.4.2\vendor\bundle\jruby\2.5.0\gems\i18n-1.7.0\lib\i18n\core_ext\string\interpolate.rb


Alternatively the source it may be looking for is from an older version of the i18n gem... you can find it here:
https://github.com/ruby-i18n/i18n/tree/0.6/lib/i18n/core_ext/string

Everything appears to be working by adding either a blank file or the old gem's interpolate.rb file.  Your mileage may vary of course, we've only used this workaround in a development environment and have no familiarity with this plug-ins source code.

**Comment by michael-doubez** [comment]
It seems to be related to ruby version support:
https://github.com/ruby-i18n/i18n/issues/299

**Comment by michael-doubez** [comment]
#89
